ui/src/app/machines/views/MachineDetails/MachineNetwork/BridgeForm/BridgeForm.tsx

```diff
@@ -15,14 +55,94 @@ const BridgeForm = ({
   selected,
   systemId,
 }: Props): JSX.Element | null => {
+  const dispatch = useDispatch();
+  const machine = useSelector((state: RootState) =>
+    machineSelectors.getById(state, systemId)
+  );
+  const cleanup = useCallback(() => machineActions.cleanup(), []);
+  const nextName = getNextNicName(machine, NetworkInterfaceTypes.BRIDGE);
+  const [{ linkId, nicId }] = selected;
+  const nic = useSelector((state: RootState) =>
+    machineSelectors.getInterfaceById(state, systemId, nicId, linkId)
+  );
+  const vlan = useSelector((state: RootState) =>
+    vlanSelectors.getById(state, nic?.vlan_id)
+  );
+  const vlansLoading = useSelector(vlanSelectors.loading);
+  const { errors, saved, saving } = useMachineDetailsForm(
+    systemId,
+    "creatingBridge",
+    "createBridge",
+    () => close()
+  );
+
+  useEffect(() => {
+    dispatch(vlanActions.fetch());
+  }, [dispatch]);
+
   // A bridge can only be created with one interface.
   if (selected.length !== 1) {
     return null;
   }
+
+  if (vlansLoading || !nic || !machine || !("interfaces" in machine)) {
+    return <Spinner text="Loading..." />;
+  }
+
   return (
     <FormCard sidebar={false} stacked title="Create bridge">
       <InterfaceFormTable interfaces={selected} systemId={systemId} />
-      <button onClick={close}>Cancel</button>
+      <FormikForm
+        buttons={FormCardButtons}
+        cleanup={cleanup}
+        errors={errors}
+        initialValues={{
+          ...networkFieldsInitialValues,
+          bridge_fd: 15,
```

Comment:
Is this value meaningless if `bridge_stp` is false? If so It might be safer to only set it if `bridge_stp` is set to true. I'm sure the api handles it properly but I think it's good to only set the parameters that are required.

Reply:
Done.